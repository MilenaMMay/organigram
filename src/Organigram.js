import React, { Component } from 'react'
import PropTypes from 'prop-types'

import organigramStyles from './Organigram.styles'

export default class Organigram extends Component {
  static propTypes = {
    text: PropTypes.string,
    data: PropTypes.object,
  }

  render() {
    const {
      text,
      data,
    } = this.props

    return (
      <div>
        {data.views.map((view) =>
          <div style={organigramStyles} key={view.name}>
            {view.name}
          </div>
          )}
      </div>
    )
  }
}
