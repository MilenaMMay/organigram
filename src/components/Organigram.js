import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Tabs from '@material-ui/core/Tabs'
import Tab from '@material-ui/core/Tab'
import View from './View'

export default class Organigram extends Component {
  static propTypes = {
    data: PropTypes.object,
  }

  constructor(props) {
    super(props);
    this.state = {
      data: props.data,
      selectedView: props.data.views[0]
    }
  }

  render() {
    return (
      <div>
        <Tabs
          value={this.state.selectedView.name}
          onChange={this.handleChange}
          indicatorColor="primary"
          textColor="primary"
          centered
        >
          {this.state.data.views.map((view) =>
            <Tab
              label={view.name}
              value={view.name}
              key={view.name}
              onClick={() => this.setState({selectedView: view})}
            />
          )}
        </Tabs>
        { this.state.selectedView ? <View view={this.state.selectedView}></View> : null }
      </div>
    )
  }
}
