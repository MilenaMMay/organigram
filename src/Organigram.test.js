import React from 'react';
import Organigram from './Organigram'
import renderer from 'react-test-renderer';

test('Organigram renders views', () => {
  const component = renderer.create(
    <Organigram data={{ views: [ { name: 'Foo' }, { name: 'Bar' } ] }}></Organigram>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
