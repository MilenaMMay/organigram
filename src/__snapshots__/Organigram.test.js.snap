// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organigram renders views 1`] = `
<div>
  <div
    style={
      Object {
        "border": "2px solid #000",
        "display": "inline-block",
        "font-size": "2em",
        "margin": "2em",
        "padding": "1em",
      }
    }
  >
    Foo
  </div>
  <div
    style={
      Object {
        "border": "2px solid #000",
        "display": "inline-block",
        "font-size": "2em",
        "margin": "2em",
        "padding": "1em",
      }
    }
  >
    Bar
  </div>
</div>
`;
